plugins/kubernetes/app/models/watchers/deploy_watcher.rb

```diff
@@ -0,0 +1,83 @@
+require 'celluloid'
+require 'celluloid/autostart'
+
+module Watchers
+  # Instantiated when a Kubernetes deploy is created to watch the status
+  class DeployWatcher
+    include Celluloid
+    include Celluloid::Notifications
+
+    finalizer :on_termination
+
+    def initialize(release)
+      @release = release
+      @current_rcs = {}
+      Rails.logger.info "Start watching K8s deploy: #{@release}"
```

Comment:
On my side of things I include `Celluloid::Internals::Logger` and use that one (see https://github.com/celluloid/celluloid/wiki/Logging). It would be good to agree how and which loggers do we use. If we decide to go with Rails logger, we would benefit from plugging it into Celluloid to have traces of actors crashing. Currently I'm creating a new log file when the plugin starts: `Celluloid.logger = ::Logger.new(File.join(Rails.root, 'log', 'celluloid.log'))`